pkg/mimirpb/custom_test.go

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/cortexproject/cortex/blob/master/pkg/cortexpb/compat_test.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Cortex Authors.
+
+package mimirpb
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/prometheus/prometheus/model/histogram"
+)
+
+func TestIsFloatHistogram(t *testing.T) {
+	tests := map[string]struct {
+		histogram Histogram
+		expected  bool
+	}{
+		"empty int histogram": {
+			histogram: FromHistogramToHistogramProto(0, &histogram.Histogram{
+				Sum:           0,
+				Count:         0,
+				ZeroThreshold: 0.001,
+				Schema:        2,
+				PositiveSpans: []histogram.Span{
+					{Offset: 0, Length: 1},
+					{Offset: 3, Length: 1},
+					{Offset: 2, Length: 2},
+				},
+				PositiveBuckets: []int64{0, 0, 0, 0},
```

Comment:
This is an invalid histogram. We should not be supporting this, but rather fixing the source of such histograms. This fix can possibly create more subtle and hard-to-debug bugs. TSDB won't ingest these samples even if given. (in a way, you found out that the count was 0 because of the code assuming it as a float, so it was a good thing IMO that surfaced a bug in histogram generation)

Reply:
Ok my bad, I did not see the contents of the bucket, it is indeed valid